packages/start/src/client/createMiddleware.ts

```diff
@@ -152,67 +152,107 @@ export interface MiddlewareOptions<
   >
 }
 
+export type MiddlewareServerNextFn = <
+  TNewServerContext = undefined,
+  TNewClientAfterContext = undefined,
+>(ctx?: {
+  context?: TNewServerContext
+  sendContext?: DefaultTransformerStringify<TNewClientAfterContext>
+}) => Promise<
+  ServerResultWithContext<TNewServerContext, TNewClientAfterContext>
+>
+
+export interface MiddlewareServerFnOptions<
```

Comment:
Separating the options into separate types made a large difference with instantiations. My guess is there is more cachine going on